woocommerce/woocommerce-gutenberg-products-block
# Make Filter Products by Stock block compatible with PHP rendered Classic Template block

This PR updates the Filter Products By Stock block to work with the Classic Template block which is PHP rendered, but also still maintains compatibility with the All Products block.

Filter Products by Stock block will only work with the PHP rendered Classic Template block if the filters are applied to the URL and the page is re-rendered. This behavior happens with the Classic Themes such as Storefront.




Fixes #6137 

## Technical details

I believe there is a lot of space to improve the code regarding all the blocks that filter the products.
Now, there are a lot of side effects, hard to debug, and these blocks are not performant (there are a lot of rerender :'( )

 Also, [As said by ](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6138#issuecomment-1098868583) @dinhtungdu this approach is not really extendible when we will blockify this template.

My suggestion is to dedicate some time (in cooldown or in a dedicated project) to improve the quality of the codebase regarding the block that handles filters cc @Aljullu 

### Testing

#### Automated Tests
 
### Manual Testing

How to test the changes in this Pull Request:

1. With a block theme, go to Appearance > Editor > Template > Product Catalog.
2. Add the Filter Products by Stock block above the product grid.
3. Be sure that you have some out stock products. Check "out of stock" on the frontend.
4. Notice the page reloads and the list of products updates accordingly.
5. Add a test page to your site with the `All Products` block, and the `Filter Product by Stock` block and ensure no regressions have been introduced here.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).


### Changelog

> Allow adding the Filter Products by Stock block to Product Catalog templates to filter products.
